src/ngx_http_lua_module.c

```diff
@@ -661,6 +679,64 @@ ngx_http_lua_init(ngx_conf_t *cf)
         lmcf->postponed_to_access_phase_end = 0;
     }
 
+    len = lmcf->lua_cpath.len;
+
+    if (lmcf->requires_append_lua_path) {
+        for (i = 0; i < lmcf->append_lua_cpath->nelts; i++) {
+            p = (ngx_str_t *) lmcf->append_lua_cpath->elts;
+            len += p[i].len;
+        }
+
+        path = (char *) ngx_palloc(cf->pool, len);
+        if (path == NULL) {
+            return NGX_ERROR;
+        }
+
+        ngx_memzero(path, len);
+
+        for (i = 0; i < lmcf->append_lua_cpath->nelts; i++) {
+            p = (ngx_str_t *) lmcf->append_lua_cpath->elts;
+            path = strncat(path, (char *) p[i].data, p[i].len);
+        }
+
+        if (lmcf->lua_cpath.len) {
+            path = strncat(path, (char *) lmcf->lua_cpath.data,
+                           lmcf->lua_cpath.len);
+        }
+
+        lmcf->lua_cpath.data = (u_char *) path;
+        lmcf->lua_cpath.len = len;
+    }
+
+    len = lmcf->lua_path.len;
+
+    if (lmcf->requires_append_lua_path) {
+        for (i = 0; i < lmcf->append_lua_path->nelts; i++) {
+            p = (ngx_str_t *) lmcf->append_lua_path->elts;
+            len += p[i].len;
+        }
+
+        path = (char *) ngx_palloc(cf->pool, len);
+        if (path == NULL) {
+            return NGX_ERROR;
+        }
+
+        ngx_memzero(path, len);
+
+        for (i = 0; i < lmcf->append_lua_path->nelts; i++) {
+            p = (ngx_str_t *) lmcf->append_lua_path->elts;
+            path = strncat(path, (char *) p[i].data, p[i].len);
+        }
+
+        if (lmcf->lua_path.len) {
+            path = strncat(path, (char *) lmcf->lua_path.data,
```

Comment:
Can we just use `ngx_copy()` instead to do the concatenation? It's more of the nginx style.

Please fix other similar places.